modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java

```diff
@@ -472,6 +515,149 @@ protected void deactivate() {
 		}
 	}
 
+	private SearchContext _buildSearchContext(
+		long companyId, String keywords, int start, int end, Sort sort) {
+
+		SearchContext searchContext = new SearchContext();
+
+		searchContext.setAttributes(
+			HashMapBuilder.<String, Serializable>put(
+				Field.CODE, keywords
+			).put(
+				Field.ENTRY_CLASS_PK, keywords
+			).put(
+				Field.NAME, keywords
+			).put(
+				"params",
+				LinkedHashMapBuilder.<String, Object>put(
+					"keywords", keywords
+				).build()
+			).build());
+		searchContext.setCompanyId(companyId);
+		searchContext.setEnd(end);
+
+		if (Validator.isNotNull(keywords)) {
+			searchContext.setKeywords(keywords);
+		}
+
+		if (sort != null) {
+			searchContext.setSorts(sort);
+		}
+
+		searchContext.setStart(start);
+
+		QueryConfig queryConfig = searchContext.getQueryConfig();
+
+		queryConfig.setHighlightEnabled(false);
+		queryConfig.setScoreEnabled(false);
+
+		return searchContext;
+	}
+
+	private List<CommerceCurrency> _getCommerceCurrencies(Hits hits)
+		throws PortalException {
+
+		List<Document> documents = hits.toList();
+
+		List<CommerceCurrency> commerceCurrencies = new ArrayList<>(
+			documents.size());
+
+		for (Document document : documents) {
+			long commerceCurrencyId = GetterUtil.getLong(
+				document.get(Field.ENTRY_CLASS_PK));
+
+			CommerceCurrency commerceCurrency = fetchCommerceCurrency(
+				commerceCurrencyId);
+
+			if (commerceCurrency == null) {
+				Indexer<CommerceCurrency> indexer =
+					IndexerRegistryUtil.getIndexer(CommerceCurrency.class);
+
+				long companyId = GetterUtil.getLong(
+					document.get(Field.COMPANY_ID));
+
+				indexer.delete(companyId, document.getUID());
+			}
+			else if (commerceCurrency != null) {
+				commerceCurrencies.add(commerceCurrency);
+			}
+		}
+
+		return commerceCurrencies;
+	}
+
+	private List<CommerceCurrency> _getCommerceCurrencies(
+			Hits hits, boolean navigationActive)
+		throws PortalException {
+
+		List<Document> documents = hits.toList();
+
+		List<CommerceCurrency> commerceCurrencies = new ArrayList<>(
+			documents.size());
+
+		for (Document document : documents) {
+			long commerceCurrencyId = GetterUtil.getLong(
+				document.get(Field.ENTRY_CLASS_PK));
+
+			CommerceCurrency commerceCurrency = fetchCommerceCurrency(
+				commerceCurrencyId);
+
+			if (commerceCurrency == null) {
+				Indexer<CommerceCurrency> indexer =
+					IndexerRegistryUtil.getIndexer(CommerceCurrency.class);
+
+				long companyId = GetterUtil.getLong(
+					document.get(Field.COMPANY_ID));
+
+				indexer.delete(companyId, document.getUID());
+			}
+			else if ((commerceCurrency != null) &&
+					 (commerceCurrency.isActive() == navigationActive)) {
```

Comment:
I don't get this, I think this can be put in some prefilter/seachContext clause to filter out unwanted results